### PR TITLE
split kernel

### PR DESCRIPTION
**Pull request by BelehovEgor**
move cache to new nuget

### CODE REVIEW

File: Kernel.Redis/src/Kernel.Redis/Helpers/CacheNotebook.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ public Frame(int database, string key, TimeSpan lifeTime)
 
     public void Add(List<Guid> elementsIds, int database, string key)
     {
-      foreach (Guid elementId in elementsIds)
+      foreach (var elementId in elementsIds)
```
Comment (MariyaSin) [inline Kernel.Redis/src/Kernel.Redis/Helpers/CacheNotebook.cs]: why do you change to var?

File: Kernel.Redis/src/Kernel.Redis/Helpers/CacheNotebook.cs [hunk 2]
```diff
@@ -68,12 +68,12 @@ public void Add(Guid elementId, int database, string key)
 
     public async Task RemoveAsync(Guid elementId)
     {
-      if (!_dictionary.TryRemove(elementId, out List<Frame> frames) || frames == null)
+      if (!_dictionary.TryRemove(elementId, out var frames) || frames == null)
```
Comment (MariyaSin) [inline Kernel.Redis/src/Kernel.Redis/Helpers/CacheNotebook.cs]: same

File: Kernel.Redis/src/Kernel.Redis/Helpers/CacheNotebook.cs [hunk 3]
```diff
@@ -68,12 +68,12 @@ public void Add(Guid elementId, int database, string key)
...
       {
         return;
       }
 
-      foreach (Frame frame in frames.Where(f => !f.IsOverdue))
+      foreach (var frame in frames.Where(f => !f.IsOverdue))
```
Comment (MariyaSin) [inline Kernel.Redis/src/Kernel.Redis/Helpers/CacheNotebook.cs]: same

File: Kernel.Redis/src/Kernel.Redis/Helpers/RedisHelper.cs [hunk 4]
```diff
@@ -30,11 +30,11 @@ public async Task CreateAsync<T>(int database, string key, T item, TimeSpan? lif
 
     public async Task<T> GetAsync<T>(int database, string key)
     {
-      RedisValue projectsFromCache = await _cache.GetDatabase(database).StringGetAsync(key);
+      var projectsFromCache = await _cache.GetDatabase(database).StringGetAsync(key);
```
Comment (MariyaSin) [inline Kernel.Redis/src/Kernel.Redis/Helpers/RedisHelper.cs]: same

File: Kernel.Redis/src/Kernel.Redis/Helpers/RedisHelper.cs [hunk 5]
```diff
@@ -30,11 +30,11 @@ public async Task CreateAsync<T>(int database, string key, T item, TimeSpan? lif
...
 
       if (projectsFromCache.HasValue)
       {
-        T item = JsonConvert.DeserializeObject<T>(projectsFromCache);
+        var item = JsonConvert.DeserializeObject<T>(projectsFromCache);
```
Comment (MariyaSin) [inline Kernel.Redis/src/Kernel.Redis/Helpers/RedisHelper.cs]: same

**Review by MariyaSin — APPROVED** [review]

**Review by Cheephot — APPROVED** [review]

File: Kernel.Redis/src/Kernel.Redis/Helpers/RedisHelper.cs (hunk 5 above)
Comment (BelehovEgor) [reply Kernel.Redis/src/Kernel.Redis/Helpers/RedisHelper.cs]: fixed

File: Kernel.Redis/src/Kernel.Redis/Helpers/RedisHelper.cs (hunk 4 above)
Comment (BelehovEgor) [reply Kernel.Redis/src/Kernel.Redis/Helpers/RedisHelper.cs]: fixed

File: Kernel.Redis/src/Kernel.Redis/Helpers/CacheNotebook.cs (hunk 3 above)
Comment (BelehovEgor) [reply Kernel.Redis/src/Kernel.Redis/Helpers/CacheNotebook.cs]: fixed

File: Kernel.Redis/src/Kernel.Redis/Helpers/CacheNotebook.cs (hunk 2 above)
Comment (BelehovEgor) [reply Kernel.Redis/src/Kernel.Redis/Helpers/CacheNotebook.cs]: fixed

File: Kernel.Redis/src/Kernel.Redis/Helpers/CacheNotebook.cs (hunk 1 above)
Comment (BelehovEgor) [reply Kernel.Redis/src/Kernel.Redis/Helpers/CacheNotebook.cs]: fixed

**Review by MariyaSin — APPROVED** [review]

**Review by Cheephot — APPROVED** [review]

**Review by VLiamin — APPROVED** [review]

**Pull request merged**